phd-tests/framework/src/test_vm/mod.rs

```diff
@@ -226,18 +230,53 @@ impl TestVm {
             migrate,
         };
 
-        let mut retries = 3;
-        while let Err(e) =
-            self.client.instance_spec_ensure().body(&ensure_req).send().await
-        {
-            info!("Error {} while creating instance, will retry", e);
-            tokio::time::sleep(Duration::from_millis(500)).await;
-            retries -= 1;
-            if retries == 0 {
-                tracing::error!("Failed to create instance after 3 retries");
-                anyhow::bail!(e);
-            }
-        }
+        // There is a brief period where the Propolis server process has begun
+        // to run but hasn't started its Dropshot server yet. Ensure requests
+        // that land in that window will fail, so retry them. This shouldn't
+        // ever take more than a couple of seconds (if it does, that should be
+        // considered a bug impacting VM startup times).
+        backoff::future::retry(
+            backoff::ExponentialBackoff {
+                max_elapsed_time: Some(std::time::Duration::from_secs(2)),
+                ..Default::default()
+            },
+            || async {
+                if let Err(e) = self
+                    .client
+                    .instance_spec_ensure()
+                    .body(&ensure_req)
+                    .send()
+                    .await
+                {
+                    match e {
+                        propolis_client::Error::CommunicationError(e) => {
+                            info!(%e,
+                                  "retriable error from instance_spec_ensure");
+                            Err(backoff::Error::Transient {
+                                err: anyhow!(
+                                    "retriable error from instance_spec_ensure \
+                                    ({:?})",
+                                    e
+                                ),
+                                retry_after: None,
+                            })
```

Comment:
style nit, take it or leave it:: there's a convenience constructor for this in `backoff`:
```suggestion
                            Err(backoff::Error::transient(anyhow!(
                                "retriable error from instance_spec_ensure ({e:?})",
                            ))
```

Reply:
also, we might prefer to use `anyhow::Context`, rather than creating a new error using `anyhow!`:
```suggestion
                            Err(backoff::Error::transient(
                                e.context("retriable error from instance_spec_ensure")
                            ))
```

Reply:
I started using the convenience constructor in 84f94e93. I'm not sure about how to do the `context` bit, though--IIUC `anyhow` implements that for `Result`, not for the errors themselves. I tried throwing an extra `.context` onto the preceding call to `instance_spec_ensure` and then downcasting the error back to a `propolis_client::Error` to try to figure out whether it was transient or not, but the downcast panicked for reasons I'd need to investigate.

Because we're logging a message in either case (and we don't really care about the innards of the error in the transient case anyway), in 4ff11a67 I dropped the construction of the extra `anyhow` error and just put the Propolis client error directly into the backoff error.

Reply:
cool, seems fine to me --- I *thought* there was also a `context` method for errors, but maybe I hallucinated that, ChatGPT-style :)